### ISSUE TITLE
[Feature Idea] Default to wildcard if too many merged imports

### ISSUE THREAD

**Issue by nasadorian** [liancheng/scalafix-organize-imports#19]
If I have a number of qualified imports sharing a package name, such as below:

```scala
import cats.data.AndThen
import cats.data.Binested
import cats.data.BinestedBitraverse 
import cats.data.Kleisli
import cats.data.Reader
import cats.data.OptionT
import cats.data.State
import cats.data.StateT
```

The optimizer under the Merge configuration (rightly) groups them all together:

```scala
import cats.data.{AndThen, Binested, BinestedBitraverse, Kleisli, OptionT, Reader, State, StateT}
```

However at some point the number of things imported from the same package can be so large that the merge grouping will go over the column limit. Of course, Scalafmt can catch this and turn it into this:

```scala
import cats.data.{
  AndThen,
  Binested,
  BinestedBitraverse,
  Kleisli,
  Reader,
  OptionT,
  State,
  StateT
}
```

But why not have an option to simplify to a wildcard like below? It's much cleaner, albeit less precise w.r.t scope.

```scala
import cats.data._
```

So the feature idea I'm proposing is to add a configuration to the `groupedImports` rule such as `SmartMerge` which will default to a wildcard if the merged group goes over the standard column limit (100 chars?).

**Comment by liancheng** [comment]
Thanks for the idea! IntelliJ import organizer supports this feature, and I actually implemented but removed it later because it may introduce correctness issues. For example:

- Unwanted implicits may get imported by the wildcard import
- The wildcard import might bring in conflict names:

  ```scala
  import scala.collection._
  import scala.collection.mutable._

  val m: Map[Int, Int] = ???
  ```

  Assuming that the `scala.collection.mutable._` import was introduced by `OrganizeImports` when coalescing multiple `scala.collection.mutable` imports, now the complier cannot resolve which `Map` should be used for `m`.

But maybe we can still have this feature and disable it by default. Users can use it for their own risks.

**Comment by liancheng** [comment]
BTW, the reason why the `groupedImports` option defaults to `Explode` is that having each import selector on their own line as a fully-qualified import is friendlier to version control and text searching. I personally prefer not grouping them together. It also avoids exceeding the column limit.

**Comment by nasadorian** [comment]
Yup, I got the idea from IntelliJ's optimizer! The scope clobbering side effect is definitely a real concern, especially when implicits are involved.

Maybe this config should be named DangerMerge instead of SmartMerge.

Does your implementation exist somewhere on a branch in this repo? I'm curious to see it.

**Comment by liancheng** [comment]
It was introduced in https://github.com/liancheng/scalafix-organize-imports/commit/f027392fee8e354cb14c71baa528c170e09ce109, and later removed in https://github.com/liancheng/scalafix-organize-imports/commit/013ba08b7411587938224a0f1cfd628713505252.

**Comment by liancheng** [comment]
@nasadorian, this feature is now included in v0.3.0-RC1.

**Comment by nasadorian** [comment]
Awesome, thanks for the update @liancheng! Gonna take it for a spin in one of my repos.
